railgun/test/integration/ingress_bulk_test.go

```diff
@@ -0,0 +1,145 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/annotations"
+	"github.com/kong/kubernetes-testing-framework/pkg/utils/kubernetes/generators"
+)
+
+const (
+	maxBatchSize             = 50
+	testBulkIngressNamespace = "ingress-bulk-testing"
+)
+
+// TestIngressBulk attempts to validate functionality at scale by rapidly deploying a large number of ingress resources.
+func TestIngressBulk(t *testing.T) {
+	ctx := context.Background()
+
+	t.Logf("creating namespace %s for testing", testBulkIngressNamespace)
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testBulkIngressNamespace}}
+	namespace, err := env.Cluster().Client().CoreV1().Namespaces().Create(ctx, namespace, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up namespace %s", testBulkIngressNamespace)
+		require.NoError(t, env.Cluster().Client().CoreV1().Namespaces().Delete(ctx, namespace.Name, metav1.DeleteOptions{}))
```

Comment:
So does deleteNamespace, we should move them to pkg/util, so that old and new version can be leveraged.

Reply:
As per our slack conversation, we've got a follow up item to eventually coordinate having only one reusable mechanism for namespace management: https://github.com/Kong/kubernetes-ingress-controller/issues/1574